src/scheduler/fifo.c

```diff
@@ -564,9 +566,64 @@ schedule(int cmd, int sd, char *runjobid)
 				"reconfigure", "Scheduler is reconfiguring");
 			free_fairshare_head(conf.fairshare);
 			reset_global_resource_ptrs();
+
+			if (conf.prime_sort != NULL) {
+				for (i = 0; i < MAX_SORTS; i++) {
+					if (conf.prime_sort[i].res_name != NULL) {
+						free(conf.prime_sort[i].res_name);
+						if (conf.prime_sort[i].res_name != conf.non_prime_sort[i].res_name) {
+							conf.prime_sort[i].res_name = NULL;
+						} else {
+							conf.prime_sort[i].res_name = conf.non_prime_sort[i].res_name = NULL;
+						}
+					}
+				}
+			}
 			free(conf.prime_sort);
+
+			if (conf.non_prime_sort != NULL) {
+				for (i = 0; i < MAX_SORTS; i++) {
+					if (conf.non_prime_sort[i].res_name != NULL)
+						free(conf.non_prime_sort[i].res_name);
+				}
+			}
 			free(conf.non_prime_sort);
 
+			if (conf.prime_node_sort != NULL) {
+				for (i = 0; i < MAX_SORTS; i++) {
```

Comment:
Make a helper function for this loop.  You do it 4 times.

Reply:
Created a new function named free_sort_info() and used it appropriately.